### PR TITLE
refactor: clean up several files

### PR DESCRIPTION
**Pull request by alan-agius4**
Most of these changes are requires for TS 4 update

### CODE REVIEW

File: packages/ngtools/webpack/src/transformers/make_transform.ts [hunk 1]
```diff
@@ -99,8 +99,9 @@ export function makeTransform(
 // If TS sees an empty decorator array, it will still emit a `__decorate` call.
 //    This seems to be a TS bug.
 function cleanupDecorators(node: ts.Node) {
-  if (node.decorators && node.decorators.length === 0) {
-    node.decorators = undefined;
+  if (node.decorators?.length === 0) {
+    // tslint:disable-next-line:no-any
+    (node as any).decorators = undefined;
```
Comment (clydin) [inline packages/ngtools/webpack/src/transformers/make_transform.ts:104]: I'm curious if this workaround is still needed at all now.

File: packages/angular_devkit/schematics_cli/bin/schematics.ts [hunk 2]
```diff
@@ -267,7 +267,8 @@ export async function main({
       return argv._;
     }
   });
-  delete parsedArgs._;
+
+  parsedArgs._  = [];
```
Comment (clydin) [inline packages/angular_devkit/schematics_cli/bin/schematics.ts:271]: The behavior is now different.  Does this have any effect on usages of the property?

Comment (alan-agius4) [reply packages/angular_devkit/schematics_cli/bin/schematics.ts:271]: This property is marked as non optional thus it cannot be deleted.

From the schematics it seems that an undefined or empty array have the same behaviour.

File: packages/ngtools/webpack/src/transformers/make_transform.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/ngtools/webpack/src/transformers/make_transform.ts:104]: Can take a look when doing the upgrade.

Comment (alan-agius4) [reply packages/ngtools/webpack/src/transformers/make_transform.ts:104]: @clydin just checked this, and it seems that the issue is still present.

Expected
```
import { __decorate } from "tslib"; import { AnotherDecorator } from 'another-lib'; export class AppComponent { constructor() { this.title = 'app'; } onEscape() { console.log('run'); } } __decorate([ AnotherDecorator() ], AppComponent.prototype, "onEscape", null); AppComponent = __decorate([], AppComponent)
```
to equal
```
import { __decorate } from "tslib"; import { AnotherDecorator } from 'another-lib'; export class AppComponent { constructor() { this.title = 'app'; } onEscape() { console.log('run'); } } __decorate([ AnotherDecorator() ], AppComponent.prototype, "onEscape", null);'.
```

**Pull request merged**